src/index.ts

```diff
@@ -72,39 +77,67 @@ export async function transpile(
   solcInput: SolcInput,
   solcOutput: SolcOutput,
   paths: Paths,
-  options?: TranspileOptions,
+  options: TranspileOptions = {},
 ): Promise<OutputFile[]> {
+  const nodeData: { node: Node; data: Partial<TransformData> }[] = [];
+
   const outputPaths = getExtraOutputPaths(paths, options);
-  const alreadyInitializable = findAlreadyInitializable(solcOutput, options?.initializablePath);
+  const alreadyInitializable = findAlreadyInitializable(solcOutput, options.initializablePath);
 
   const excludeSet = new Set([...alreadyInitializable, ...Object.values(outputPaths)]);
-  const excludeMatch = matcher(options?.exclude ?? []);
+  const peerExcludedSet = new Set<string>();
+  const excludeMatch = matcher(options.exclude ?? []);
 
   const namespaceInclude = (source: string) => {
-    const namespaced = options?.namespaced ?? false;
-    const namespaceExclude = options?.namespaceExclude ?? [];
+    const namespaced = options.namespaced ?? false;
+    const namespaceExclude = options.namespaceExclude ?? [];
     return namespaced && !namespaceExclude.some(p => minimatch(source, p));
   };
 
+  if (options.peerProject !== undefined) {
+    for (const [source, { ast }] of Object.entries(solcOutput.sources)) {
+      const importFromPeer = path.join(options.peerProject, source);
+      const contracts = [...findAll('ContractDefinition', ast)];
+
+      // populate nodeData
+      nodeData.push(
+        ...contracts
+          .filter(node => node.contractKind !== 'contract')
+          .map(node => ({ node, data: { importFromPeer } })),
+      );
+
+      // soft exclude files that have no
+      if (contracts.every(({ contractKind }) => contractKind !== 'contract')) {
+        peerExcludedSet.add(source);
+      }
+    }
+  }
+
   const transform = new Transform(solcInput, solcOutput, {
-    exclude: source => excludeSet.has(source) || (excludeMatch(source) ?? isRenamed(source)),
+    exclude: source =>
+      peerExcludedSet.has(source)
+        ? 'soft'
+        : excludeSet.has(source) || (excludeMatch(source) ?? isRenamed(source)),
```

Comment:
what should take priority between `true` and `soft`. I wrote it that way because its easier, but maybe we want to do 
```suggestion
    exclude: source =>
      excludeSet.has(source) || (excludeMatch(source) ?? isRenamed(source))
        ? true
        : peerExcludedSet.has(source)
        ? 'soft'
        : false,
```

Reply:
"Hard" should take priority, as the most conservative option.

Reply:
I implemented `true` taking priority over `soft`.

Should we do `boolean | 'soft'` or `'hard' | 'soft' | false` ?

Reply:
My preference is for `'hard' | 'soft' | false`

Reply:
done